src/Auth0.php

```diff
@@ -664,13 +661,13 @@ public function decodeIdToken(string $idToken, array $verifierOptions = []) : ar
         }
 
         $verifierOptions = $verifierOptions + [
-            // Set a custom leeway if one was passed to the constructor.
             'leeway' => $this->idTokenLeeway,
+        // Set a custom leeway if one was passed to the constructor.
             'max_age' => $this->transientHandler->getOnce('max_age') ?? $this->maxAge,
+            self::TRANSIENT_NONCE_KEY => $this->transientHandler->getOnce(self::TRANSIENT_NONCE_KEY)
         ];
 
-        $verifierOptions[self::TRANSIENT_NONCE_KEY] = $this->transientHandler->getOnce(self::TRANSIENT_NONCE_KEY);
```

Comment:
Moved this up to allow passed-in verifier options `nonce` value to remain (in the case where `false` is passed in)

Reply:
The `nonce` check should be optional, only when present. e.g. when set by the authentication starter method. 
We still want to set it by default whenever the authentication is started. But in the case of a refresh token exchange, there will never be a nonce claim back in the ID token to check.

Instead of opening up the public API with a param to override the nonce value, encapsulate this internals here by not failing when it's not set.

Reply:
Removed the failure below. ID Token Validator will (still) accept an empty nonce value and skip that check.